src/main/java/com/exadel/bookService/controller/BookController.java

```diff
@@ -0,0 +1,45 @@
+package com.exadel.bookService.controller;
+
+import com.exadel.bookService.exception.BookNotFoundException;
+import com.exadel.bookService.model.Book;
+import com.exadel.bookService.service.BookService;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/books")
+public class BookController {
+
+    private final BookService service;
+
+    public BookController(BookService service) {
+        this.service = service;
+    }
+
+    @GetMapping
+    public List<Book> getAllBooks() {
+        return service.getAllBooks();
+    }
+
+    @GetMapping("/{id}")
+    public Book getBookById(@PathVariable Long id) {
+        return service.getBookById(id)
+                .orElseThrow(() -> new BookNotFoundException("Book with id " + id + " not found"));
```

Comment:
Throwing exceptions must be on business level but not in controller. Please move it into the BookService